src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -66,31 +66,33 @@ getLastLedgerSequence(STTx const& tx)
 }
 
 static
-std::uint64_t
-increase(std::uint64_t level,
+FeeLevel64
+increase(FeeLevel64 level,
     std::uint32_t increasePercent)
 {
-    return mulDiv(
-        level, 100 + increasePercent, 100).second;
+    return mulDiv(level, 100 + increasePercent, 100).second;
 }
 
 
 //////////////////////////////////////////////////////////////////////////
 
+constexpr FeeLevel64 TxQ::baseLevel;
+
 std::size_t
 TxQ::FeeMetrics::update(Application& app,
     ReadView const& view, bool timeLeap,
     TxQ::Setup const& setup)
 {
-    std::vector<uint64_t> feeLevels;
+    std::vector<FeeLevel64> feeLevels;
     auto const txBegin = view.txs.begin();
     auto const txEnd = view.txs.end();
     auto const size = std::distance(txBegin, txEnd);
     feeLevels.reserve(size);
     std::for_each(txBegin, txEnd,
         [&](auto const& tx)
         {
-            auto const baseFee = calculateBaseFee(view, *tx.first);
+            auto const baseFee = view.fees().toDrops(
+                calculateBaseFee(view, *tx.first)).second;
```

Comment:
This looks like a change in behavior, is that intentional?

Reply:
Yes, absolutely. This is one of the places that I identified as treating the value returned as a different unit than it was intended to be. In practice, right now, the numeric value is going to be equivalent, so there will be no behavior change, but if the base fee ever changes, that will be significant.